src/cli.rs

```diff
@@ -60,6 +61,13 @@ pub struct PackageSelectOptions {
     /// regardless, set this flag.
     #[structopt(long)]
     ignore_publish: bool,
+    /// Automatically detect the packages, which changed compared to the given git commit.
+    ///
+    /// Compares the current git `head` to the reference given, identifies which files changed
+    /// and attempts to identify the packages through that mechanism. You can use any
```

Comment:
I removed original mention of identifying dependent crates here. It seems that doing specifically that is required mostly for semver auto bump but if we want to, let's say, add owners to every package that was changed since some git ref than we shouldn't really perform dependency analysis - very similar to --packages.

Because it, I think we should implement dependency analysis as a complement to either `--packages` or `--changed-since` as a separate flag (e.g. `--with-dependents=true/false`) for publish/semver-related subcommands.